packages/headless/src/api/analytics/insight-analytics.ts

```diff
@@ -72,6 +72,15 @@ export class InsightAnalyticsProvider
     return buildFacetStateMetadata(getStateNeededForFacetMetadata(this.state));
   }
 
+  public getGeneratedAnswerMetadata() {
+    const state = this.getState();
+    const formattedObject: Record<string, string | boolean> = {};
+    if (state.generatedAnswer?.isVisible !== undefined) {
+      formattedObject['showGeneratedAnswer'] = state.generatedAnswer.isVisible;
+    }
+    return formattedObject;
+  }
```

Comment:
I think this would also do ?

```suggestion
public getGeneratedAnswerMetadata() {
    const state = this.getState();
    return {
      ...(state.generatedAnswer?.isVisible !== undefined && {
        showGeneratedAnswer: state.generatedAnswer.isVisible,
      }),
    };
}
```

Reply:
Exactly 👌